tutorials/basic-topics/linked-list.md

```diff
@@ -1,10 +1,688 @@
+
+<Tabs
+    defaultValue="java"
+    values={[
+        {label: 'Java', value: 'java'},
+    ]}>
+<TabItem value="java">
+
+```java
+Node temp = head;
+  // we are running the loop from 2 node because we have to move to the node 
+  // just before the node we want to remove
+  for(int i=2; i<pos; i++){
+    // change pointer from current node to the next node
+    temp = temp.next;
+  }
+temp.next = temp.next.next;
+```
+</TabItem>
+</Tabs>
+
+**Leetcode 237. Delete Node in a Linked List:** 
+
+There is a singly linked list. You have to delete a $node$ from the list. You are given the $node$ to delete but not the $head$ of the list. Delete the given $node$. Note that by deleting the $node$, we $do not$ mean removing it from memory. We mean:
+
+*The value of the given node should not exist in the linked list.
+*The number of nodes in the linked list should decrease by one.
+*All the values before node should be in the same order.
+*All the values after node should be in the same order.
+
+**Approach:** Change the value of the $current node$ to the value of the $next$ node. Do this until the last node.
+
+**Solution:**
+
+<Tabs
+    defaultValue="java"
+    values={[
+        {label: 'Java', value: 'java'},
+    ]}>
+<TabItem value="java">
+
+```java
+class Solution {
+    // we are given a node to delete
+    public void deleteNode(ListNode node) {
+        // check if the current node is second last node
+        while(node.next.next!=null){
+            // change the value of current node to the value of next node
+            node.val = node.next.val;
+            // move node to the next node
+            node=node.next;
+        }
+
+        // we are at the second last node
+        // change value of the last node to the value of next node
+        node.val = node.next.val;
+        // change the next of last node to null
+        node.next = null;
+    }
+}
+``` 
+</TabItem>
+</Tabs>
+
+## Search for an item in the list
+
+* Make a temporary pointer $ptr$ point to $head$
+* Move $ptr$ to next node until $ptr$ is $null$
+* At each iteration, check if the data in $ptr$ is same as the number we want. If yes, then return $true$
+* Return $false$ if number not found
+
+<Tabs
+    defaultValue="java"
+    values={[
+        {label: 'Java', value: 'java'},
+    ]}>
+<TabItem value="java">
+
+```java
+Node ptr = head;  
+while(ptr!=null){
+  // check if data in current node matches the number we are looking for
+  if(ptr.data == num){  
+    // return true if number found
+    return true;   
+  }
+  // move ptr to the next node
+  ptr = ptr.next;  
+}
+// return false if number not found
+return false;  
+``` 
+</TabItem>
+</Tabs>
+
+## Let's take a look at some Leetcode problems:
+
+**Leetcode 19. Remove Nth Node From End of List**
+
+Given the head of a linked list, remove the nth node from the end of the list and return its head. Given Linked list: $[1] -> [2] -> [3] -> [4] -> [5]$ for $n=2$ change the list to $[1] -> [2] -> [3] ->     [5]$.
```

Comment:
looks weird in the app.

Reply:
just this one or everywhere in the app. I can replace it with an image.

Reply:
Use the following

```
Given the head of a linked list, remove the nth node from the end of the list and return its head. Given Linked list: $[1]$ -> $[2]$ -> $[3]$ -> $[4]$ -> $[5]$ for $n = 2$ change the list to $[1]$ -> $[2]$ -> $[3]$ -> $[5]$.
```